pennylane/operation.py

```diff
@@ -300,48 +300,149 @@ def hash(self):
         """int: returns an integer hash uniquely representing the operator"""
         return hash((str(self.name), tuple(self.wires.tolist()), _process_data(self)))
 
+    # pylint:disable=unused-argument
+    @staticmethod
+    def _matrix(*params, **hyperparams):
+        """Base matrix for the matrix representation of this operator.
+
+        The base matrix assumes the wire order of the operator's wires.
+
+        Args:
+            params (Iterable): trainable parameters that influence the matrix representation
+            hyperparams (dict): other inputs that influence the matrix representation
+        Returns:
+            tensor-like: base matrix, or None if representation is not defined
+        """
+        return None
+
     @classmethod
-    def _matrix(cls, *params):
+    def compute_matrix(cls, *params, wires=None, wire_order=None, **hyperparams):
```

Comment:
I'm just wondering if it makes sense to take `params` as a single iterable, instead of an unpacked arbitrary sized list.  Then the other variables can be passed as arguments instead of limiting them to only keyword arguments: `op.compute_matrix(op.parameters, op.wires, wire_order, op.hyperparameters)`.

Reply:
@mariaschuld from reading this, it looks like it makes a lot of sense to simply 'merge' `_matrix` and `compute_matrix`, and make `compute_matrix` a static method?